lib/archethic/transaction_chain.ex

```diff
@@ -970,8 +970,15 @@ defmodule Archethic.TransactionChain do
           | {:error, :not_in_local}
   def resolve_paging_state(_, nil, _), do: {:ok, nil}
 
-  def resolve_paging_state(_, paging_state, _) when is_binary(paging_state),
-    do: {:ok, paging_state}
+  def resolve_paging_state(last_chain_address, paging_state, _) when is_binary(paging_state) do
+    genesis_address = get_genesis_address(last_chain_address)
```

Comment:
The node using this function may not be a storage node of this chain so the genesis address may not be retrieved.
Maybe the best place do do this check is in the `ChainReader.get_transaction_chain` function